### ISSUE TITLE
[How to] Relate contacts to project milestones or project tasks?

### ISSUE THREAD

**Issue by Hammerman3000** [YetiForceCompany/YetiForceCRM#8897]
Hey there,

I am having the following use case:
Let's say a vendor contact has a certain qualification. So I want this person to be part of a realization, however, the qualification allows this contact only to be part of a certain task or milestone and not the complete project. How would I do that?

The reason is simple: Projects would become too chaotic quickly if contacts are assigned on a "general" project level instead of milestone or task level.

How do I do that? :)

Cheers,
Hamfred

**Comment by Hammerman3000** [comment]
No one with any idea? 
Is this a missing feature or simply a mistake by myself not to see the way how to do it?

**Comment by TomaszQr** [comment]
You can add relation M2M between milestone and contacts
![relation_1](https://user-images.githubusercontent.com/14273211/50593112-3e532e00-0e97-11e9-9ed1-b2df84b5cbac.png)
after you should see this tab
![relation_2](https://user-images.githubusercontent.com/14273211/50593120-5034d100-0e97-11e9-9d23-0a3a56f9da75.png)

**Comment by Hammerman3000** [comment]
Well, i miss that function it seems...
Running on Yetiforce 4.3

![image](https://user-images.githubusercontent.com/45795852/50605412-86d41100-0ec2-11e9-8855-f4698533a3e4.png)

**Comment by mariuszkrzaczkowski** [comment]
https://github.com/YetiForceCompany/YetiForceCRM/blob/17b91310d7c473e842e60b79dc29fa03998b3ccd/config/developer.php#L17-L18

Set true

**Comment by Hammerman3000** [comment]
Works, thank you very much @mariuszkrzaczkowski!
I just think i discovered a bug... I was able to replicate it twice:

1) I am linking Project Milestones and Contacts 
2) I am linking Contacts and Project Milestones
3) Now I create a Project "Demo 1234" and a Milestone "First Milestone" and within the milestone select a contact to be assigned to it
4) Checking at the contact Tab where I made the fields "Project Milestone" "Project" "assigned to" "Status" and "last modified" appear is working all fine
5) Checking in the project milestone shows the correct contact
6) Now I rename the Project to "Demo 5678" and the milestone to "First Milestone of Project Demo 5678"
7) I go back in the milestone and things are fine, I see the assigned contact
8) I check at the contact and yetiforce would actually manage to "freeze" while loading / load forever

--> I deleted all projects and created new ones, did not solve the issue of loading forever
--> I rebooted the server, run all CRONs, did not solve the issue
--> I deleted the link at the "Contacts" side towards "Project Milestones" and re-established it --> This solved the issue immediately
--> I went over to rename a project again and repeat the steps above = freeze
--> Deleting relation and re-establishing it = solved

So I think there is an issue that maybe the project name or project milestone name are hardcoded into to contact with their label instead of a relative ID or something?
Is this possible?
